packages/@ama-sdk/core/src/fwk/param-deserialization.spec.ts

```
@@ -0,0 +1,69 @@
+import {
+  deserializePathParams,
+  deserializeQueryParams,
+} from './param-deserialization';
+
+describe('Deserialize parameters', () => {
+  it('should correctly deserialize query parameters of type primitive', () => {
+    expect(deserializeQueryParams({ idPrimitive: 'idPrimitive=5' }, { idPrimitive: { explode: true, style: 'form', paramType: 'primitive' } })).toEqual({ idPrimitive: '5' });
+    expect(deserializeQueryParams({ idPrimitive: 'idPrimitive=5' }, { idPrimitive: { explode: false, style: 'form', paramType: 'primitive' } })).toEqual({ idPrimitive: '5' });
+  });
+
+  it('should correctly deserialize query parameters of type array', () => {
+    expect(deserializeQueryParams({ idArray: 'idArray=3&idArray=4&idArray=5' }, { idArray: { explode: true, style: 'form', paramType: 'array' } })).toEqual({ idArray: ['3', '4', '5'] });
+    expect(deserializeQueryParams({ idArray: 'idArray=3,4,5' }, { idArray: { explode: false, style: 'form', paramType: 'array' } })).toEqual({ idArray: ['3', '4', '5'] });
+    expect(deserializeQueryParams({ idArray: 'should-not-work' }, { idArray: { explode: true, style: 'spaceDelimited', paramType: 'array' } })).toEqual({ idArray: undefined });
```

should this return `undefined` or throw an error?

---

this should return `undefined`, we don't throw errors for non-supported use cases

---

I would expect a function like this to throw when the format is not recognized, but I'm ok to return `undefined`

---

**Update:** we modified in order to throw errors for non-supported use cases